aspnetcore/hosting/ihostingstartup.md

```diff
@@ -0,0 +1,162 @@
+
+```
+<DRIVE>\Users\<USER>\.dotnet\store\x64\netcoreapp2.0\<FEATURE_ASSEMBLY_NAME>\<FEATURE_VERSION>\lib\netcoreapp2.0\
+```
+
+For global use, place the assembly in the .NET Core installation's runtime store:
+
+```
+<DRIVE>\Program Files\dotnet\store\x64\netcoreapp2.0\<FEATURE_ASSEMBLY_NAME>\<FEATURE_VERSION>\lib\netcoreapp2.0\
+```
+
+When deploying the assembly to the runtime store, the symbols file may be deployed as well but isn't required for the feature to work.
+
+**Place the dependencies file**
+
+The implementation's *\*.deps.json* file must be in an accessible location.
+
+For per-user use, place the file in the `additonalDeps` folder of the user profile's `.dotnet` settings: 
+
+```
+<DRIVE>\Users\<USER>\.dotnet\x64\additionalDeps\<FEATURE_ASSEMBLY_NAME>\shared\Microsoft.NETCore.App\2.0.0\
+```
+
+For global use, place the file in the `additonalDeps` folder of the .NET Core installation:
+
+```
+<DRIVE>\Program Files\dotnet\additionalDeps\<FEATURE_ASSEMBLY_NAME>\shared\Microsoft.NETCore.App\2.0.0\
+```
+
+Note the version, `2.0.0`, reflects the version of the shared runtime that the target app uses. The shared runtime is shown in the *\*.runtimeconfig.json* file. In the sample app, the shared runtime is specified in the *HostingStartupSample.runtimeconfig.json* file.
+
+**Set environment variables**
+
+Set the following environment variables in the context of the app that uses the feature.
+
+ASPNETCORE\_HOSTINGSTARTUPASSEMBLIES
+
+Only hosting startup assemblies are scanned for the `HostingStartupAttribute`. The assembly name of the implementation is provided in this environment variable. The sample app sets this value to `StartupDiagnostics`.
+
+The value can also be set using the [Hosting Startup Assemblies](xref:fundamentals/hosting#hosting-startup-assemblies) host configuration setting.
+
+DOTNET\_ADDITIONAL\_DEPS
+
+The location of the implementation's *\*.deps.json* file.
+
+If the file is placed in the user profile's *.dotnet* folder for per-user use:
+
+```
+<DRIVE>\Users\<USER>\.dotnet\x64\additionalDeps\
+```
+
+If the file is placed in the .NET Core installation for global use, provide the full path to the file:
+
+```
+<DRIVE>\Program Files\dotnet\additionalDeps\<FEATURE_ASSEMBLY_NAME>\shared\Microsoft.NETCore.App\2.0.0\<FEATURE_ASSEMBLY_NAME>.deps.json
+```
+
+The sample app sets this value to:
+
+```
+%UserProfile%\.dotnet\x64\additionalDeps\StartupDiagnostics\
+```
+
+For examples of how to set environment variables for various operating systems, see [Working with multiple environments](xref:fundamentals/environments).
+
+## Discover loaded hosting startup assemblies
+
+To discover hosting startup assemblies loaded by the app or libraries added to the app, read the value of the [HostingStartupAssembliesKey](/dotnet/api/microsoft.aspnetcore.hosting.webhostdefaults.hostingstartupassemblieskey). The sample app reads the key into a `string` array and displays the result in the app's Index page:
```

Comment:
First check the application logs:
https://github.com/aspnet/Hosting/blob/86df76440b450daa3972d5b317026b58f718b803/src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs#L136-L151